File: IntegrationSerial.c
#include <stdio.h>
#include <math.h>
#include <time.h>

int main(){
    float integral;
    float a,b;
    int n;
    float h;
    float x;
    clock_t start,end;
    double cpu_time_used;


    float f(float x);
    printf("Enter a, b and n \n");
    scanf("%f %f %d", &a, &b, &n);
    start = clock();

    h = (b-a)/n;
    integral = (f(a) + f(b))/2.0;
    x = a;
    for( int i = 1; i <= n-1; i++){
        x += h;
        integral += f(x);
    }
    integral *= h;
    end = clock();
    cpu_time_used = ((double) (end - start)) / CLOCKS_PER_SEC;
    printf(" With n = %d trapezoids, our estimate \n", n);
    printf("of the integral form %f to %f = %f\n", a, b, integral);
    printf("Program took %f seconds to execute \n", cpu_time_used);

}

float f(float x){ 
    return x*x;
}

File: IntegrationSerial.py
import matplotlib.pyplot as plt
import time

a = float(input("Enter Value for upper limit: \n"))
b = float(input("Enter vale for lower limit: \n"))
n = int(input("Enter number of trapezoids to form: \n"))

num1 = int(a)
num2 = int(b)
begin = time.time()
def fx(x):
    return x*x

height = (b-a)/n
integral = (fx(a) + fx(b))/2.0
x = a
for i in range(1,n):
    x += height
    integral += fx(x)


integral *= height
end = time.time()
totaltime = end - begin
print(" With n = %d trapezoids, our estimate \n" %(n))
print("of the integral from %f to %f = %0.8f\n" %(a,b,integral))
print("Total runtime of the program is %f" %(totaltime))

xcords = range(num1,num2)
ycords = [x*x for x in xcords]
plt.plot(xcords,ycords)
plt.show()
